lib/index.js

```diff
@@ -83,224 +83,237 @@ If not version is specified or wildcard (['*']), it would be enought with <contr
 
 */
 
-var Rest = function(settings) {
-  this.settings = {};
-  this.routes = [];
-
-  settings.versioning = settings.versioning || {};
-  this.vnd_header = settings.versioning.header ? settings.versioning.header.toLowerCase() : 'accept';
-  //Example: /vnd.mycompany.com\+json; version=(\d+)(,|$)/;
-  this.vnd_grab = settings.versioning.grab;
-  this.vnd_error = settings.versioning.error || 406;
-
-  delete settings.versioning;
-
-  if(settings) {
-    var self = this;
-    _.each(settings, function(value, key) {self.settings[key] = value; });
+export default class Rest {
+  constructor(settings) {
+    this.settings = {};
+    this.routes = [];
+    this.nestedPath = null;
+    this.subfolder = null;
+
+    settings.versioning = settings.versioning || {};
+    this.vnd_header = settings.versioning.header?.toLowerCase() || 'accept';
+    // Example: /vnd.mycompany.com\+json; version=(\d+)(,|$)/;
+    this.vnd_grab = settings.versioning.grab;
+    this.vnd_error = settings.versioning.error || 406;
+
+    delete settings.versioning;
+    this.settings = { ...settings };
```

Comment:
```suggestion
    this.settings = {...settings};
    this.routes = [];
    this.nestedPath = null;
    this.subfolder = null;

    this.vnd_header = settings?.versioning?.header?.toLowerCase() || 'accept';
    // Example: /vnd.mycompany.com\+json; version=(\d+)(,|$)/;
    this.vnd_grab = settings?.versioning?.grab;
    this.vnd_error = settings?.versioning?.error || 406;
```

Reply:
Make sense?

Reply:
may be necessary to `delete this.settings.versioning`

Reply:
To maintain the old behavior.
The previous version extracted things from the settings.versioning into this, and then deleted that prop from settings.

Afaik the only setting in use is this.settings.controllers, so it shouldn't hurt to leave it as you said.

Another option would be to not save the settings object, but to change the behavior to do this.base_dir = settings.controllers, which is the only current use for the settings.

Reply:
What do you think about:
```js
  constructor(settings = {}) {
    this.routes = [];
    this.nestedPath = null;
    this.subfolder = null;

    this.base_dir = settings.controllers;

    const versioning = settings.versioning || {};
    this.vnd_header = versioning.header?.toLowerCase() || 'accept';
    // Example: /vnd.mycompany.com\+json; version=(\d+)(,|$)/;
    this.vnd_grab = versioning.grab;
    this.vnd_error = versioning.error || 406;
  }
```

Reply:
Coolllll